modules/core/src/action.ts

```diff
@@ -20,11 +21,31 @@ const hasOwnProperty = <P extends string | number>(
   return typeof obj === 'object' && obj !== null && obj.hasOwnProperty(prop)
 }
 
+abstract class BaseAction {
```

Comment:
Would recommend changing the nomenclature to `EmptyAction` in the way that we have ComponentNext.empty() which has certain behaviour. Basically then `Action` will be a proper monoid and we could potentially build concats and reduces over actions later.

Reply:
atleast, I think `EmptyAction` would be semantically correct based on our discussion the other day; `Action.nil` is right now acting like a `never` type (which nothing can extend). So we need something to act like an `unknown`, which anything can extend in the future. `Empty` seems like it would fulfil that role.